### ISSUE TITLE
Issue of generating adversarial samples in the demo

### ISSUE THREAD

**Issue by liuzihao7543** [cyhuang-tw/attack-vc#1]
Dear authors,

Thanks for providing the high quality codes and the nice paper. I am trying to generate the adversarial samples in the demo. Since it is not a targeted attack(no y in the demo), I could not use the targeted attack code you provide to generate the adversarial samples directly. Could you please provide the untargeted end-to-end attack  code (for equation (1) in the paper )for generating the demo samples? 

My attempted modification of the e2e_attack function for untargeted attack is:

def e2e_attack(
    model: nn.Module,
    vc_src: Tensor,
    vc_tgt: Tensor,
    adv_tgt: Tensor,
    eps: float,
    n_iters,
) -> Tensor:
    ptb = torch.zeros_like(vc_tgt).normal_(0, 1).requires_grad_(True)
    opt = torch.optim.Adam([ptb])
    criterion = nn.MSELoss()
    pbar = trange(n_iters)
    

    with torch.no_grad():
        org_out = model.inference(vc_src, vc_tgt)
      
    losses = []
    ptbs=[]

    for _ in pbar:
        adv_inp = vc_tgt + ptb
        adv_out = model.inference(vc_src, adv_inp)
        loss =  -  criterion(adv_out, org_out)
        opt.zero_grad()
        loss.backward()
        opt.step()
        ptb.data.clamp_(-eps, eps)
 
    return vc_tgt + ptb

However, the adversarial samples I generate(.e.g epsilon = 0.1 ) is different from yours in the demo.
Any advise or reply will be highly appreciated!

Zihao

**Comment by cyhuang-tw** [comment]
Hi,

We used targeted attack in the demo and experiments.
You can use untargeted attack by removing ```criterion(adv_out, tgt_out)``` in ```attack_utils.py```, line 27, and your code seems correct.

Initialization is important in adversarial attacks. Different initializations may lead to very different perturbations.
I think it is normal if you obtained a different sample (perturbation) from the demo page.

**Comment by liuzihao7543** [comment]
Thank you so much for answering my questions. That makes sense.
